src/rgw/rgw_data_sync.cc

```diff
@@ -1726,18 +1743,13 @@ class RGWDataSyncShardCR : public RGWCoroutine {
             spawn(sync_single_entry(source_bs, log_iter->entry.key, log_iter->log_id,
                                     log_iter->log_timestamp, false), false);
           }
-          while ((int)num_spawned() > spawn_window) {
-            set_status() << "num_spawned() > spawn_window";
-            yield wait_for_child();
-            int ret;
-            while (collect(&ret, lease_stack.get())) {
-              if (ret < 0) {
-                tn->log(10, "a sync operation returned error");
-                /* we have reported this error */
-              }
-              /* not waiting for child here */
-            }
-          }
+
+          drain_all_but_stack_cb(lease_stack.get(),
```

Comment:
yikes, data sync was respecting the `spawn_window` here before, but with this change it blocks after spawning each bucket sync

this PR added a `yield_spawn_window()` for bucket sync, so i opened https://github.com/ceph/ceph/pull/45567 to use that here in data sync too